### ISSUE TITLE
fast-beam-search support for sherpa-onnx

### ISSUE THREAD

**Issue by uni-manjunath-ke** [k2-fsa/sherpa-onnx#347]
I could see fast-beam-search decoding is currently not supported in sherpa-onnx. Is this activity is planned for future? If yes, when can this be expected (timeline)? In specific, do you plan to support fast-beam-search-with-lg.

Because, we could see that fast-beam-search-with-lg is not supported in zipformer/streaming-decode.py. 
Thanks

**Comment by csukuangfj** [comment]
`fast_beam_search` depends on k2. If you want to use fast_beam_search, please use k2-fsa/sherap.

 We are going to support HLG decoding in sherpa-onnx. If you want to use a lexicon or/and an n-gram LM in decoding, then you may find HLG decoding interesting.

Please also have a look at https://github.com/k2-fsa/icefall/pull/1275

**Comment by uni-manjunath-ke** [comment]
Sure, Thanks. Ya, we are interested in HLG. Hoping to see this HLG code at https://github.com/k2-fsa/icefall/pull/1275 merged soon and available for usage.

**Comment by uni-manjunath-ke** [comment]
Hi @csukuangfj , 
Just wanted to confirm, if the HLG code at [https://github.com/k2-fsa/icefall/pull/1275](https://github.com/k2-fsa/icefall/pull/1275 ) is only for CTC,  or  it can be used with zipformers as well. Could you please confirm. Because, we are interested in Zipformer models for now.

**Comment by uni-manjunath-ke** [comment]
Currently, I see in https://github.com/k2-fsa/icefall/pull/1275 that HLG support is added for ICefall, When can we expect this support to be ported for sherpa-onnx?  or Is that already ported to Sherpa-onnx? Thanks

**Comment by csukuangfj** [comment]
> Currently, I see in [k2-fsa/icefall#1275](https://github.com/k2-fsa/icefall/pull/1275) that HLG support is added for ICefall, When can we expect this support to be ported for sherpa-onnx? or Is that already ported to Sherpa-onnx? Thanks

please see https://github.com/k2-fsa/sherpa-onnx/pull/349

We will finish it in two weeks.

---

> if the HLG code at https://github.com/k2-fsa/icefall/pull/1275 is only for CTC

Yes, you are right. It is only for CTC.

---

> it can be used with zipformers as well

Zipformer is a kind of neural network, while ctc is a kind of loss function. They are two different things.
If you train a zipformer using CTC loss, then you can use HLG decoding with zipformer.
If you train a zipformer using transducer loss, then you cannot use HLG decoding with zipformer.

**Comment by csukuangfj** [comment]
> Currently, I see in [k2-fsa/icefall#1275](https://github.com/k2-fsa/icefall/pull/1275) that HLG support is added for ICefall, When can we expect this support to be ported for sherpa-onnx? or Is that already ported to Sherpa-onnx? Thanks

@uni-manjunath-ke 

https://github.com/k2-fsa/sherpa-onnx/pull/349

The C++ part is usable now.

**Comment by uni-manjunath-ke** [comment]
Thank you.

**Comment by programindz** [comment]
> Zipformer is a kind of neural network, while ctc is a kind of loss function. They are two different things. If you train a zipformer using CTC loss, then you can use HLG decoding with zipformer. If you train a zipformer using transducer loss, then you cannot use HLG decoding with zipformer.

For transducer type of models, what kind of decoding is available

**Comment by csukuangfj** [comment]
@dosu
